amxmodx/amxmodx.cpp

```diff
@@ -1722,26 +1722,36 @@ static cell AMX_NATIVE_CALL get_concmdsnum(AMX *amx, cell *params) /* 1 param */
 	return g_commands.getCmdNum(CMD_ConsoleCommand, params[1]);
 }
 
-static cell AMX_NATIVE_CALL register_event(AMX *amx, cell *params) /* 2 param */
+// native register_event(const event[], const function[], const flags[], const cond[] = "", ...);
+static cell AMX_NATIVE_CALL register_event(AMX *amx, cell *params)
 {
 	CPluginMngr::CPlugin* plugin = g_plugins.findPluginFast(amx);
 
-	int len, pos, iFunction;
+	int len, evenId, iFunction;
 
-	char* sTemp = get_amxstring(amx, params[1], 0, len);
+	const char* eventName = get_amxstring(amx, params[1], 0, len);
 
-	if ((pos = g_events.getEventId(sTemp)) == 0)
+	if ((evenId = g_events.getEventId(eventName)) == 0)
 	{
-		LogError(amx, AMX_ERR_NATIVE, "Invalid event (name \"%s\") (plugin \"%s\")", sTemp, plugin->getName());
+		LogError(amx, AMX_ERR_NATIVE, "Invalid event (name \"%s\") (plugin \"%s\")", eventName, plugin->getName());
 		return 0;
 	}
 
-	sTemp = get_amxstring(amx, params[2], 0, len);
-	iFunction = registerSPForwardByName(amx, sTemp, FP_CELL, FP_DONE);
+	const char* callback = get_amxstring(amx, params[2], 0, len);
+
+	auto event = g_events.findEvent(plugin, evenId, callback);
+
+	if (event)
```

Comment:
This will cause bcompat issues and I think we should go with the Ham-style enable/disable API.